flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTableImpl.java

```diff
@@ -89,6 +97,9 @@
     /** Mapping from job id to allocated slots for a job. */
     private final Map<JobID, Set<AllocationID>> slotsPerJob;
 
+    /** Working directory base to write allocation files */
+    private final File allocationsDirectory;
```

Comment:
I am wondering whether the persistence logic should be part of the `TaskSlotTableImpl`. The reason is that it adds another concern to this class. Moreover, all other `TaskSlotTable` implementations won't have this feature then. Alternatively, we could either introduce a decorator for a `TaskSlotTable` or do the persistence logic outside of the `TaskSlotTable`.